tests/pytests/test_issues.py

```diff
@@ -822,127 +822,60 @@ def test_mod5791(env):
-                    node_with_slot_0_port = int(k.split(":")[1])
-                    break
-
-            env.assertIsNotNone(node_with_slot_0_port)
-            new_shard_id = new_instance_conn.cluster_myid(cluster.ClusterNode('127.0.0.1', new_instance_port))
-            conn.cluster_setslot(cluster.ClusterNode('127.0.0.1', node_with_slot_0_port), new_shard_id, 0, 'NODE')
-            new_instance_conn.cluster_setslot(cluster.ClusterNode('127.0.0.1', new_instance_port), new_shard_id, 0, 'NODE')
-
-            # Validate the updated state in old and new shards.
-            wait_for_expected(lambda: get_node_by_port(conn, new_instance_port)['slots'], [['0']],
-                              'waiting for cluster slots to update')
-            wait_for_expected(lambda: get_node_by_port(new_instance_conn, new_instance_port)['slots'], [['0']],
-                              'waiting for cluster slots to update')
-            break  # test succeeded - we can finish
-        except Exception as e:
-            if attempt == num_retries-1:  # last attempt failed - probably a real problem - raise the exception
-                raise e
-            cleanup_required = False  # no need to clean up - we are rerunning
-            env.debugPrint(f"Exception caught: {str(e)} - retrying")
-
-        finally:
-            # cleanup
-            if cleanup_required:
-                new_instance.kill()
-                os.remove('nodes.conf')
+    conn = env.getConnection()
+    initial_shards_count = env.shardsCount
+    # The first two fields in the cluster info reply are the number of partition in thr cluster.
+    env.assertEqual(conn.execute_command("search.clusterinfo")[:2], ['num_partitions', int(initial_shards_count)])
+
+    # Add a new shard to the cluster. Internally we call CLUSTER MEET to connect the new shard
+    # to the cluster. Also, we internally wait for the cluster to be ready and call "search.CLUSTERREFRESH"
+    # and update the topology change in the new shard (this is where we had a crash in MOD-5778).
+    env.addShardToClusterIfExists()
+    new_shard_conn = env.getConnection(shardId=initial_shards_count+1)
+    # Expect that the cluster will be aware of the new shard, but for redisearch coordinator, the new shard doesn't
+    # consider part of the partition yet as it does not contain any slot.
+    env.assertEqual(int(new_shard_conn.execute_command("cluster info")['cluster_known_nodes']), initial_shards_count+1)
+    env.assertEqual(new_shard_conn.execute_command("search.clusterinfo")[:2], ['num_partitions', int(initial_shards_count)])
+
+    # Move one slot (0) to the new shard (according to https://redis.io/commands/cluster-setslot/)
+    new_shard_id = new_shard_conn.execute_command('CLUSTER MYID')
+    env.assertEqual(new_shard_conn.execute_command(f"CLUSTER SETSLOT 0 NODE {new_shard_id}"), "OK")
+    env.assertEqual(conn.execute_command(f"CLUSTER SETSLOT 0 NODE {new_shard_id}"), "OK")
```

Comment:
why twice?

Reply:
This is how you are supposed to do that - notify the new shard and then the old shard (see https://redis.io/commands/cluster-setslot)